src/resolve/resolvectl.c

```diff
@@ -1930,28 +1930,32 @@ static int verb_dns(int argc, char **argv, void *userdata) {
         if (r < 0)
                 return bus_log_create_error(r);
 
-        STRV_FOREACH(p, argv + 2) {
-                struct in_addr_data data;
+        /* If only argument is the empty string, then call SetLinkDNS() with an
+         * empty list, which will clear the list of domains for an interface. */
+        if (!(argc == 3 && isempty(argv[2]))) {
```

Comment:
I prefer to use `strv_equal()` and `STRV_MAKE()` here, that is, `if (!strv_equal(argv + 2, STRV_MAKE("")))`.